code/xcvm/lib/core/src/network.rs

```diff
@@ -26,14 +35,28 @@ pub struct UserOrigin {
 	pub user_id: UserId,
 }
 
+impl Display for UserOrigin {
+	#[inline]
+	fn fmt(&self, fmtr: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+		core::write!(fmtr, "{}-{}", self.network_id, self.user_id)
+	}
+}
+
 /// Arbitrary `User` type that represent the identity of a user on a given network, usually a public
 /// key.
-#[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
+#[serde_as]
 #[derive(
```

Comment:
same here about json schema in public api

Reply:
Implemented below on line 176.